rdap_types/src/lib.rs

```diff
@@ -123,28 +120,30 @@ impl<'de> Deserialize<'de> for CountryCode {
 }
 
 /// https://tools.ietf.org/html/rfc7483#section-4.2
-#[skip_serializing_none]
 #[derive(Serialize, Deserialize, Debug)]
 pub struct Link {
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub value: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub rel: Option<String>,
     pub href: String,
-    pub hreflang: Option<Vec<String>>,
+    #[serde(rename = "hreflang", skip_serializing_if = "Option::is_none")]
+    pub href_lang: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub title: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub media: Option<String>,
-    #[serde(rename = "type")]
-    pub typ: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub r#type: Option<String>,
 }
 
 /// Value signifying the relationship an object would have with its closest containing object.
 /// Values come from [RFC 7483] and [RDAP JSON Values].
 ///
 /// [RFC 7483]: https://tools.ietf.org/html/rfc7483#section-10.2.4
 /// [RDAP JSON Values]: https://www.iana.org/assignments/rdap-json-values/rdap-json-values.xhtml
-#[skip_serializing_none]
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Copy)]
-#[serde(rename_all = "lowercase")]
-#[serde(remote = "Role")]
+#[serde(rename_all = "lowercase", remote = "Role")]
```

Comment:
I assume, the `remote` here exists to be able to implement the De/Serialize traits manually. But, the behavior of `.tirol` is actually a spec violation. If we want to maintain this, we need to figure out how we want to support these cases.
Maybe have all registered roles here and then a `Unregistered(String)` variant?

Based on: https://rdap.ryce-rsp.com/rdap/domain/nic.tirol
.tirol claims to implement both "icann_rdap_technical_implementation_guide_0" and "icann_rdap_response_profile_0". 
The technical guide says under 4.2:
> If the server responses include roles which are not listed below, such roles MUST
be registered at the IANA RDAP JSON Values Registry..

I checked the registry and `tech` and `admin` are not registered.

Reply:
Replaced non-standard variants with an `Unknown(String)` variant.